agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4363,6 +4365,45 @@ bool slave_thread::handle_monitor_ap_metrics_response(Socket &sd,
     return true;
 }
 
+bool slave_thread::handle_monitor_beacon_metrics_response(Socket &sd,
+                                                          ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Forwarding BEACON_METRICS_RESPONSE_MESSAGE to controller, mid=" << std::hex
+               << int(mid);
+
+    auto monitor_beacon_response = cmdu_rx.getClass<wfa_map::tlvBeaconMetricsResponse>();
+
+    if (!monitor_beacon_response) {
+        LOG(ERROR) << "Failed getClass<wfa_map::tlvBeaconMetricsResponse>";
+        return false;
+    }
+
+    auto beacon_response =
+        cmdu_tx.create(mid, ieee1905_1::eMessageType::BEACON_METRICS_RESPONSE_MESSAGE);
+
+    if (!beacon_response) {
+        LOG(ERROR) << "Failed to create BEACON_METRICS_RESPONSE_MESSAGE";
+        return false;
+    }
+
+    auto beacon_metrics_response_tlv = cmdu_tx.addClass<wfa_map::tlvBeaconMetricsResponse>();
+
+    if (!beacon_metrics_response_tlv) {
+        LOG(ERROR) << "Failed addClass<wfa_map::tlvBeaconMetricsResponse>";
+        return false;
+    }
+
+    // TODO: Add filling up the correct data according to the specification.
```

Comment:
it seems the monitor response and the forwarded message contain the exact same TLV: `tlvBeaconMetricsResponse`.
do we, and if not - shouldn't we, support a full copy of a TLV?
@morantr might be able to answer this.